# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import statsmodels.stats.api as sms

C:\Users\USER\Anaconda3\lib\site-packages\statsmodels\compat\pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

# Answer 1

In [5]:
#What test is appropriate for this problem? Does CLT apply?

print('There are ' + str(len(data['race'])) + ' records. CLT applies because there are sufficient sample size.')
print('A two sample t-test is appropriate in this case.')

There are 4870 records. CLT applies because there are sufficient sample size.
A two sample t-test is appropriate in this case.


# Answer 2

What are the null and alternate hypotheses?

Null Hypothesis: Mean white calls = Mean black calls

Alternative Hypothesis: Mean white calls != Mean black calls

# Answer 3

In [6]:
#Compute margin of error, confidence interval, and p-value

#creates data frame for black and white
data_b = data[data['race'] == 'b']
data_w = data[data['race'] == 'w']

#calculates mean calls for black and white
mean_b = data_b['call'].mean()
mean_w = data_w['call'].mean()
print('mean black call: ' +str(round(mean_b,3)))
print('mean white call: ' +str(round(mean_w,3)))
print('mean difference (white - black): ' + str((round(mean_w,3)-round(mean_b,3))))

#calculates 95% confidence level on a mean difference of .033
cm = sms.CompareMeans(sms.DescrStatsW(data_w['call']), sms.DescrStatsW(data_b['call']))
print('95 percent chance that mean difference (white - blakc) is within: (%.3f, %.3f)' % (cm.tconfint_diff(usevar='unequal')[0], cm.tconfint_diff(usevar='unequal')[1]))

#calculates margin of error
print('Margin of error: ' + str(round(((cm.tconfint_diff(usevar='unequal')[1] - cm.tconfint_diff(usevar='unequal')[0])/2),3)))

#performs two sample t test
two_t_test = stats.ttest_ind(data_b['call'], data_w['call'])
print('\nTesting a null hypothesis that black calls and white calls are equal, we get a t-statistic is %.3f and the p-value is %.7f.' % two_t_test)


mean black call: 0.064
mean white call: 0.097
mean difference (white - black): 0.033
95 percent chance that mean difference (white - blakc) is within: (0.017, 0.047)
Margin of error: 0.015

Testing a null hypothesis that black calls and white calls are equal, we get a t-statistic is -4.115 and the p-value is 0.0000394.


# Answer 4

With a p-value of less than 1%, there is statistical significance and we can reject the null hypothesis that mean white calls are the equal to mean black calls. We are also 95% confident that mean whit calls are between 0.017 and 0.047 greater than black mean calls. Although it is significant, these levels are not high enough to conclude that a name of an applicant is one the main factors affecting whether an applicant gets a call back.

# Answer 5

The evidence is not enough to suggest race is the main factor in callback success. Other factors such as educations levels, years of experience, honors etc. may explain more callback success.